tests/test_changelog.py

```
@@ -0,0 +1,79 @@
+        self.assertEqual([entry.version for entry in changelog.filter_entries(entries, "BuildKit")], ["24.0.7-1"])
+        compared = changelog.compare_entries(entries, "24.0.6-1", "24.0.7-1")
+        self.assertEqual([entry.version for entry in compared], ["24.0.7-1", "24.0.6-1"])
+
+    def test_export_json_and_cli_output(self):
+        with tempfile.TemporaryDirectory() as tempdir:
+            temp = Path(tempdir)
+            source = temp / "changelog"
+            output = temp / "out.json"
+            source.write_text(SAMPLE_CHANGELOG, encoding="utf-8")
+
+            result = subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)